src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2236,7 +2233,15 @@ private bool IsTypeDeclarationStart()
                 case SyntaxKind.ExplicitKeyword or SyntaxKind.ImplicitKeyword
                     when this.PeekToken(1).ContextualKind is SyntaxKind.ExtensionKeyword:
 
-                    return IsFeatureEnabled(MessageID.IDS_FeatureExtensions);
+                    // If we have `implicit extension E` then this is definitely an extension as it could not be parsed
+                    // as anything else.
+                    if (IsTrueIdentifier(this.PeekToken(2)))
+                        return true;
+
+                    // we have `implicit extension ...` technically this could be something like `implicit
+                    // extension.operator X(` (an implicit conversion with a explicit interface impl name). For compat,
+                    // determine the parsing strategy based on the lang version.
+                    return IsCompatBreakingFeatureEnabled(MessageID.IDS_FeatureExtensions);
```

Comment:
>IsCompatBreakingFeatureEnabled [](http://example.com/codeflow?start=27&length=30)

It doesn't look like we have this helper in main, how is it different from `IsFeatureEnabled`? If we are planning to merge this code to main, I am not certain that would be appropriate if the helper assumes we have some support for "breaking changes" feature.

Reply:
moved off of this.